app/assets/javascripts/components/students/components/Articles/SelectedStudent/RevisionsList/StudentRevisionsList.jsx

```diff
@@ -14,12 +14,26 @@ export class StudentRevisionsList extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      isOpen: false
+      isOpen: false,
+      namespace: 'all'
     };
 
     this.toggleDrawer = this.toggleDrawer.bind(this);
   }
 
+  onNamespaceChange = (e) => {
+    return this.setState({ namespace: e.target.value });
+  };
+
+  namespaceToId(nm) {
```

Comment:
This is is likely a generally useful mapping that we may want to extend later on, and may want to use in other places at some point.

I suggest extracting it to `article_utils.js`.